src/domain/flow-system/services/src/flow/flow_agent_impl.rs

```diff
@@ -729,6 +753,50 @@ impl MessageConsumerT<DatasetLifecycleMessage> for FlowAgentImpl {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+#[async_trait::async_trait]
+impl MessageConsumerT<DatasetChangedMessage> for FlowAgentImpl {
+    #[tracing::instrument(
+        level = "debug",
+        skip_all,
+        name = "FlowAgentImpl[DatasetChangedMessage]"
+    )]
+    async fn consume_message(
+        &self,
+        target_catalog: &Catalog,
+        message: &DatasetChangedMessage,
+    ) -> Result<(), InternalError> {
+        tracing::debug!(received_message = ?message, "Received dataset changed message");
+
+        let scheduling_helper = target_catalog.get_one::<FlowSchedulingHelper>().unwrap();
+        let time_source = target_catalog.get_one::<dyn SystemTimeSource>().unwrap();
+
+        match message {
+            DatasetChangedMessage::Updated(update_message) => {
+                scheduling_helper
+                    .schedule_dependent_flows(
+                        &update_message.dataset_id,
+                        DatasetFlowType::Ingest,
+                        FlowTriggerType::Push(FlowTriggerPush {
+                            trigger_time: time_source.now(),
+                            source_name: Some(FlowTriggerPushSource::HTTP),
```

Comment:
The "source_name" field was reserved to distinguish between multiple named `AddPushSource` events.